src/http/conn.rs

```diff
@@ -490,7 +497,7 @@ enum Reading {
 
 enum Writing<B> {
     Init,
-    Body(Encoder, Option<Cursor<B>>),
+    Body(Encoder, Option<Cursor<B>>, BytesMut),
```

Comment:
I hear ya, and it's a worthy goal! The best way would be to make use of `AsyncWrite::write_bufs`, which will prevent a copy in the plain-tcp-on-unix case. Unfortunately, on Windows, and when using TLS, flattening (copying) into a single buffer is required.

In this case, since we're dealing with performance sensitive stuff, I'd *love* if any changes could include benchmarks showing that it was an improvement. I've spent a lot of time twiddling with things thinking it was an obvious improvement, and it turned out to make no difference.

For extending the `write_buf`, there's the additional constraint that it should ideally never grow beyond `MAX_BUFFER_SIZE`, which could happen with APIs taking a `&mut Vec`.